extensions/markdown-language-features/src/commands/openDocumentLink.ts

```diff
@@ -83,7 +83,7 @@ export class OpenDocumentLinkCommand implements Command {
 
 	private static async tryOpen(engine: MarkdownEngine, resource: vscode.Uri, args: OpenDocumentLinkArgs, column: vscode.ViewColumn): Promise<boolean> {
 		const tryUpdateForActiveFile = async (): Promise<boolean> => {
-			if (vscode.window.activeTextEditor && isMarkdownFile(vscode.window.activeTextEditor.document)) {
+			if (vscode.window.activeTextEditor) {
```

Comment:
`tryRevealLine` will try parsing the document as markdown. Can you instead check that we only try opening non-markdown files if the fragment is a line number: 

https://github.com/microsoft/vscode/blob/c094fcf995bafb61725493fbe135492d13d6f6f5/extensions/markdown-language-features/src/commands/openDocumentLink.ts#L142